protocols/netty/src/main/java/org/apache/james/protocols/netty/EventLoopGroupManagerDefault.java

```diff
@@ -0,0 +1,128 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.protocols.netty;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.james.util.concurrent.NamedThreadFactory;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.util.concurrent.EventExecutorGroup;
+import io.netty.util.concurrent.PromiseCombiner;
+
+public final class EventLoopGroupManagerDefault implements EventLoopGroupManager {
+
+    private static final int PROCESSORS = Runtime.getRuntime().availableProcessors();
+
+    public static final int DEFAULT_BOSS_THREADS = Math.min(2, PROCESSORS);
+    public static final int DEFAULT_WORKER_THREADS = PROCESSORS * 2;
+    public static final int DEFAULT_EXECUTOR_GROUP_THREADS = PROCESSORS * 2; // TODO potentially recommend setting this to 0 so reuses worker thread pool.
```

Comment:
We can't as we have blocking logic that should not be executed on the IO eventloop...